app/services/bill-runs/two-part-tariff/prepare-licences-for-allocation.service.js

```diff
@@ -0,0 +1,142 @@
+'use strict'
+
+/**
+ * Prepares the return logs and charge elements and sorts the charge references ready for matching and allocating
+ * @module PrepareLicencesForAllocationService
+ */
+
+const DetermineAbstractionPeriodService = require('../determine-abstraction-periods.service.js')
+const DetermineChargePeriodService = require('../determine-charge-period.service.js')
+const FetchReturnLogsForLicenceService = require('./fetch-return-logs-for-licence.service.js')
+const { periodsOverlap } = require('../../../lib/general.lib.js')
+
+/**
+ * Prepares return logs and charge versions for matching and allocation based on provided licences and billing period
```

Comment:
```suggestion
 * For each licence finds returns for the billing period and prepares them and the licence's charge elements
 * ready to be matched and allocated to one another
 * 
 * > Rather than create a copy of each licence and amend it we amend the licences in place. Hence this service
 * > has no return value
```